src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1052,3 +1057,79 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
     m_ramp[deck] = true;
 }
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const ControllerScriptInterfaceLegacy::WellKnownCharsets targetCharset,
+        const QString& value) {
+    using enum WellKnownCharsets;
+    switch (targetCharset) {
+    case ASCII:
+        return convertCharsetInternal(QStringLiteral("US-ASCII"), value);
+    case UTF_8:
+        return convertCharsetInternal(QStringLiteral("UTF-8"), value);
+    case UTF_16LE:
+        return convertCharsetInternal(QStringLiteral("UTF-16LE"), value);
+    case UTF_16BE:
+        return convertCharsetInternal(QStringLiteral("UTF-16BE"), value);
+    case UTF_32LE:
+        return convertCharsetInternal(QStringLiteral("UTF-32LE"), value);
+    case UTF_32BE:
+        return convertCharsetInternal(QStringLiteral("UTF-32BE"), value);
+    case CentralEurope:
+        return convertCharsetInternal(QStringLiteral("windows-1250"), value);
+    case Cyrillic:
+        return convertCharsetInternal(QStringLiteral("windows-1251"), value);
+    case Latin1:
+        return convertCharsetInternal(QStringLiteral("windows-1252"), value);
+    case Greek:
+        return convertCharsetInternal(QStringLiteral("windows-1253"), value);
+    case Turkish:
+        return convertCharsetInternal(QStringLiteral("windows-1254"), value);
+    case Hebrew:
+        return convertCharsetInternal(QStringLiteral("windows-1255"), value);
+    case Arabic:
+        return convertCharsetInternal(QStringLiteral("windows-1256"), value);
+    case Baltic:
+        return convertCharsetInternal(QStringLiteral("windows-1257"), value);
+    case Vietnamese:
+        return convertCharsetInternal(QStringLiteral("windows-1258"), value);
+    case Latin9:
+        return convertCharsetInternal(QStringLiteral("ISO-8859-15"), value);
+    case Shift_JIS:
+        return convertCharsetInternal(QStringLiteral("Shift_JIS"), value);
+    case EUC_JP:
+        return convertCharsetInternal(QStringLiteral("EUC-JP"), value);
+    case EUC_KR:
+        return convertCharsetInternal(QStringLiteral("EUC-KR"), value);
+    case Big5_HKSCS:
+        return convertCharsetInternal(QStringLiteral("Big5-HKSCS"), value);
+    case KOI8_U:
+        return convertCharsetInternal(QStringLiteral("KOI8-U"), value);
+    case UCS2:
+        return convertCharsetInternal(QStringLiteral("ISO-10646-UCS-2"), value);
+    case SCSU:
+        return convertCharsetInternal(QStringLiteral("SCSU"), value);
+    case BOCU_1:
+        return convertCharsetInternal(QStringLiteral("BOCU-1"), value);
+    case CESU_8:
+        return convertCharsetInternal(QStringLiteral("CESU-8"), value);
+    }
+    m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
+    return QByteArray();
+}
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharsetInternal(
+        const QString& targetCharset, const QString& value) {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
```

Comment:
The 6.4 version has disappeared?

Reply:
yeah, in hindsight I there was no reason to use `QTextEncoder` when QStringEncoder is also available, so we dropped it. QStringEncoder is also available in versions older than 6.4, so we can simply use that there too.